pkg/client/results/processing.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright the Sonobuoy contributors 2019
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package results
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/heptio/sonobuoy/pkg/plugin"
+	"github.com/heptio/sonobuoy/pkg/plugin/driver/daemonset"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// StatusFailed is the key we base junit pass/failure off of and save into
+	// our canonical results format.
+	StatusFailed = "failed"
+
+	// StatusPassed is the key we base junit pass/failure off of and save into
+	// our canonical results format.
+	StatusPassed = "passed"
+
+	// StatusSkipped is the key we base junit pass/failure off of and save into
+	// our canonical results format.
+	StatusSkipped = "skipped"
+
+	// StatusUnknown is the key we fallback to in our canonical results format
+	// if another can not be determined.
+	StatusUnknown = "unknown"
+
+	// PostProcessedResultsFile is the name of the file we create when doing
+	// postprocessing on the plugin results.
+	PostProcessedResultsFile = "sonobuoy_results.yaml"
+)
+
+type postProcessor func(string) (Item, error)
+type fileSelector func(string, os.FileInfo) bool
+
+// aggregateStatus defines the aggregation rules for status. Failures bubble
+// up and otherwise the status is assumed to pass.
+func aggregateStatus(items ...Item) string {
+	for _, v := range items {
+		if len(v.Items) > 0 {
+			v.Status = aggregateStatus(v.Items...)
+		}
+		if v.Status == StatusFailed {
+			return StatusFailed
+		}
+	}
+	return StatusPassed
+}
+
+func PostProcessPlugin(p plugin.Interface, dir string) (Item, error) {
+	fmt.Printf("checking types %+v", p, p.GetResultFormat())
+	switch p.GetResultFormat() {
+	case "junit", "e2e":
+		fmt.Println("got junit item")
+		return processPluginWithProcessor(p, dir, processJunitFile, fileOrExtension(p.GetResultFile(), ".xml"))
+	case "raw":
+		fmt.Println("got raw item")
+		return processPluginWithProcessor(p, dir, processRawFile, fileOrAny(p.GetResultFile()))
+	default:
+		fmt.Println("default item?")
+
+	}
+	return Item{}, nil
+}
+
+func processPluginWithProcessor(p plugin.Interface, baseDir string, processor postProcessor, selector fileSelector) (Item, error) {
+	pdir := path.Join(baseDir, PluginsDir, p.GetName())
+	_, isDS := p.(*daemonset.Plugin)
```

Comment:
Just special handling down below for daemonsets; default to the other handling for jobs and any "other" types that could in theory exist (since the someone using this as a library could be making their own type for the plugin interface).